File: ruby/difference-of-squares/difference_of_squares.rb
class Squares
  attr_accessor :number

  def initialize(number)
    @number = number
  end

  def square_of_sum
    (0..number).sum**2
  end

  def sum_of_squares
    (0..number).inject{ |sum, number| sum + number**2}
  end

  def difference
    square_of_sum - sum_of_squares
  end
end

File: javascript/difference-of-squares/difference-of-squares.js
export class Squares {
  constructor(n) {
    this.n = n
  }

  get sumOfSquares() {
    this.sum = 0
    for (let i = 0; i < this.n + 1; i++) {this.sum += i**2}
    return this.sum
  }

  get squareOfSum() {
    return (this.n**2 + this.n)**2 / 4;
  }

  get difference() {
    return this.squareOfSum - this.sumOfSquares
  }
}
